step1_prepare/step1_1_download_data.py

```
@@ -8,9 +8,40 @@
 storybooks_dataframe.to_csv('step1_1_storybooks.csv', index=False)
 
-# Read the storybook learning events CSV into a DataFrame, and write the DataFrame to a CSV file
-storybook_learning_events_csv_url = 'https://raw.githubusercontent.com/elimu-ai/ml-datasets/refs/heads/main/lang-HIN/storybook-learning-events.csv'
-print(os.path.basename(__file__), f'storybook_learning_events_csv_url: {storybook_learning_events_csv_url}')
-storybook_learning_events_dataframe = pandas.read_csv(storybook_learning_events_csv_url)
+# Read the students CSV into a DataFrame to extract student IDs
+students_csv_url = 'https://raw.githubusercontent.com/elimu-ai/ml-datasets/main/lang-HIN/students.csv'
+print(os.path.basename(__file__), f'students_csv_url: {students_csv_url}')
+try:
+    students_dataframe = pandas.read_csv(students_csv_url)
+except Exception as e:
```

Ruff (0.15.10)

[warning] Do not catch blind exception: `Exception`

(BLE001)